Foundation/JSONSerialization.swift

```diff
@@ -823,8 +858,21 @@ private struct JSONReader {
                 defer { intEndPointer.deallocate() }
                 let doubleEndPointer = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 1)
                 defer { doubleEndPointer.deallocate() }
-                let intResult = strtol(startPointer, intEndPointer, 10)
+
+                let isNegative = startPointer.pointee == 0x2D
+                let intResult: Int64
+                let uintResult: UInt64
+
+                if isNegative {
+                    intResult = strtoll(startPointer, intEndPointer, 10)
+                    uintResult = 0
+                } else {
+                    intResult = 0
+                    uintResult = strtoull(startPointer, intEndPointer, 10)
+                }
+
```

Comment:
The `sttroll` and `strtoull` functions can both return 0 on overflow and underflow so this needs to be detected and possibly `nil` returned if the `strtod` also can't parse the number

Reply:
`strtol`, `sttroll`, `strtoull` returns max allowed value on overflow and min on underflow.
It is main problem with original code. It returns Int32.max(on android which is 32bit currently) even if number is bigger without any failing.

As I understand original code case you are described handled below
```
guard intDistance > 0 || doubleDistance > 0 else {
    return nil
}
```

Reply:
These checks aren't necessarily correct, by the way; usually the right way to validate the result of `stro*` is to check the value of where the end pointer is pointing. For instance, `strtoull("123f", intEndPointer, 10)` will return `123` and have `intEndPointer` pointing to `'f'`. Since we have a `count` and are copying into a null-terminated buffer, we should be checking the result of `intEndPointer.pointee == 0` to validate whether we parsed a full number or not.